### ISSUE TITLE
Low accuracy on small English dataset

### ISSUE THREAD

**Issue by huseinzol05** [zihangdai/xlnet#112]
I compared accuracies on small dataset, got 2 classes, which are `negative` and `positive`, each class has 5k of sentences. You can check it here [english-polarity](https://github.com/huseinzol05/NLP-Models-Tensorflow/tree/master/text-classification/data). All notebooks finetune on single Tesla V100, 100 max length of sentences. 80% to train, 20% to test. All notebooks do early stopping with 3 patients, and I will pick highest test accuracy as the benchmark.

1. [BERT English Base](https://github.com/huseinzol05/NLP-Models-Tensorflow/blob/master/text-classification/71.transfer-learning-bert-base.ipynb), test accuracy 81%.
2. [GPT2 345M](https://github.com/huseinzol05/NLP-Models-Tensorflow/blob/master/text-classification/65.transfer-learning-gpt2.ipynb), test accuracy 79%.
3. [XLnet Large](https://github.com/huseinzol05/NLP-Models-Tensorflow/blob/master/text-classification/72.xlnet-large.ipynb), test accuracy 51%.

I am not sure what is wrong here, I doubled check all the code to transform into notebooks based, maybe you can give me some guidance here, thanks. Great model and repository!

**Comment by aditya-malte** [comment]
Could you divulge the hyperparameters?

**Comment by huseinzol05** [comment]
hyperparameters are same given by all `.json` from the original repositories. I just change batch size and max length.

BERT and GPT2 doesnt have a problem when I reduce the batch and max length, still able to finetune properly.

**Comment by huseinzol05** [comment]
After I changed dropout (both for layers and attention) to 0.0, increase max length to 128, now it able to learn, testing accuracy 84%. Beat both BERT and GPT2. https://github.com/huseinzol05/NLP-Models-Tensorflow/blob/master/text-classification/72-xlnet-large.ipynb
